todo_app/tests/test_integration.py

```diff
@@ -0,0 +1,63 @@
+import os
+from dotenv import load_dotenv, find_dotenv
+import pytest
+import requests
+from todo_app import app
+from todo_app.tests.StubResponse import StubResponse
+from unittest.mock import MagicMock
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    test_app = app.create_app()
+
+    with test_app.test_client() as client:
+        yield client
+
+
+def test_index_page(monkeypatch, client):
+    monkeypatch.setattr(requests, 'get', stub_get_lists)
+
+    response = client.get('/')
+    
+    assert response.status_code == 200
+    assert 'Test card' in response.data.decode()
+
+
+def test_add_new_to_do(monkeypatch, client):
+    stub_create_to_do = MagicMock()
+    monkeypatch.setattr(requests, 'post', stub_create_to_do)
+
+    response = client.post('/', follow_redirects=True)
+
+    assert response.status_code == 200
+    assert 'To-Do App' in response.data.decode()
+    stub_create_to_do.assert_called_once_with('https://api.trello.com/1/cards', 
```

Comment:
I've opted to check the call and that the redirect to the index page works for this test. Initially I wanted to check `response.data.decode()` for evidence of the newly created card, but I couldn't quite get that to work. In the end I decided that the index endpoint was already being tested but is there a way to test this?

Reply:
There's not a great one to test them in tandem! We _can_ mock the get call after the create one, and explicitly tell it to show the new card, but that's not really giving us any more confidence than this that the underlying code works.

Pausing here and saying "The success of this endpoint depends on it calling the appropriate endpoint in Trello, and then return a 302" is a totally reasonable decision I think.